commands/create.go

```diff
@@ -540,10 +546,13 @@ func updateUserdataFile(driverOpts *rpcdriver.RPCFlags, machineName, userdataFla
 // and passes the script path to commonCloudConfig
 // RK - sets the hostname based on OS as cloud-init (linux) and cloudbase-init (windows) diverge
 // on how hostnames are set in cloud-config (userdata)
-func writeCloudConfig(machineName, encodedData, machineOS string, cf map[interface{}]interface{}, newUserDataFile *os.File) error {
+func writeCloudConfig(machineName, encodedData, machineOS, hostname string, cf map[interface{}]interface{}, newUserDataFile *os.File) error {
 	command := "sh"
 	path := "/usr/local/custom_script/install.sh"
 	key := "hostname"
+	if hostname == "" {
+		hostname = machineName
```

Comment:
You can consider it a nit, but not a fan of redefining the value of an input parameter - especially in this case where there is only a single usage on line 567 that can just be changed to
```
if hostname == "" {
	cf[key] = hostname
} else {
	cf[key] = machineName
}
```
(I miss :? operator though - https://go.dev/doc/faq#Does_Go_have_a_ternary_form)

Reply:
Updated